apps/cloud_server.c

```diff
@@ -1604,7 +1671,7 @@ main(int argc, char *argv[])
   oc_storage_config("./cloud_server_creds/");
 #endif /* OC_STORAGE */
   oc_set_factory_presets_cb(factory_presets_cb, NULL);
-  oc_set_max_app_data_size(8 * 1024 + num_resources * 512);
+  oc_set_max_app_data_size(8 * 1024 + (num_resources + 1) * 512);
```

Comment:
if we want to save up some memory then +1 should be only necessary if `OC_SOFTWARE_UPDATE` is defined